homeassistant/components/switch/command_line.py

```diff
@@ -7,31 +7,52 @@
 import logging
 import subprocess
 
-from homeassistant.components.switch import SwitchDevice
-from homeassistant.const import CONF_VALUE_TEMPLATE
+import voluptuous as vol
+
+from homeassistant.components.switch import (SwitchDevice, PLATFORM_SCHEMA)
+from homeassistant.const import (
+    CONF_FRIENDLY_NAME, CONF_SWITCHES, CONF_VALUE_TEMPLATE, CONF_COMMAND_OFF,
+    CONF_COMMAND_ON, CONF_COMMAND_STATE)
 from homeassistant.helpers import template
+import homeassistant.helpers.config_validation as cv
 
 _LOGGER = logging.getLogger(__name__)
 
+SWITCH_SCHEMA = vol.Schema({
+    vol.Optional(CONF_COMMAND_OFF, default='true'): cv.string,
+    vol.Optional(CONF_COMMAND_ON, default='true'): cv.string,
+    vol.Optional(CONF_COMMAND_STATE): cv.string,
+    vol.Optional(CONF_FRIENDLY_NAME): cv.string,
+    vol.Optional(CONF_VALUE_TEMPLATE): cv.template,
+})
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_SWITCHES): vol.Schema({cv.slug: SWITCH_SCHEMA}),
+})
 
 # pylint: disable=unused-argument
-def setup_platform(hass, config, add_devices_callback, discovery_info=None):
+def setup_platform(hass, config, add_devices, discovery_info=None):
     """Find and return switches controlled by shell commands."""
-    switches = config.get('switches', {})
-    devices = []
+    devices = config.get(CONF_SWITCHES, {})
+    switches = []
 
-    for dev_name, properties in switches.items():
-        devices.append(
+    for device_name, device_config in devices.items():
+        switches.append(
             CommandSwitch(
                 hass,
-                properties.get('name', dev_name),
-                properties.get('oncmd', 'true'),
-                properties.get('offcmd', 'true'),
-                properties.get('statecmd', False),
-                properties.get(CONF_VALUE_TEMPLATE, False)))
-
-    add_devices_callback(devices)
-
+                device_config.get(CONF_FRIENDLY_NAME, device_name),
+                device_config.get(CONF_COMMAND_ON),
```

Comment:
If `CONF_COMMAND_ON` is not a key of `device_config` it will throw a `KeyError` - this way it will return `None` if it doesn't exist